clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/DcosClientProvider.groovy

```diff
@@ -1,27 +1,32 @@
 package com.netflix.spinnaker.clouddriver.dcos
 
-import com.netflix.spectator.api.Registry
-
-import java.util.concurrent.ConcurrentHashMap
-
+import com.netflix.spinnaker.clouddriver.security.AccountCredentialsProvider
 import mesosphere.dcos.client.DCOS
 import mesosphere.dcos.client.DCOSClient
 
+import java.util.concurrent.ConcurrentHashMap
+
 class DcosClientProvider {
 
   private final Map<String, DCOS> dcosClients = new ConcurrentHashMap<>()
-  private final Registry registry
+  private final AccountCredentialsProvider credentialsProvider
 
-  DcosClientProvider(Registry registry) {
-    this.registry = registry
+  DcosClientProvider(AccountCredentialsProvider credentialsProvider) {
+    this.credentialsProvider = credentialsProvider
   }
 
-  DCOS getDcosClient(DcosCredentials account) {
-    String key = account.name
-    return dcosClients.computeIfAbsent(key, { k -> DCOSClient.getInstance(account.dcosUrl, account.dcosClientConfig) })
+  DCOS getDcosClient(DcosCredentials credentials) {
+    String key = credentials.name
+    return dcosClients.computeIfAbsent(key, { k -> DCOSClient.getInstance(credentials.dcosUrl, credentials.dcosClientConfig) })
   }
 
   DCOS getDcosClient(String accountName) {
```

Comment:
Yeah, but I don't think we can necessarily make that guarantee that the client will have already been created. It was being used in DcosJobProvider#collectJob. Since we don't really know the order in which consumers may call clouddriver endpoints, I don't think we can say it's already been created before (especially if, say, we have a read-only clouddriver instance).

Regardless, it's a public method and it didn't have any special documentation, so I think we should make it work as a consumer would expect without any preconditions.